features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseAsyncBackgroundWireframeMapper.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.mapper
+
+import android.view.View
+import com.datadog.android.sessionreplay.internal.AsyncJobStatusCallback
+import com.datadog.android.sessionreplay.internal.recorder.GlobalBounds
+import com.datadog.android.sessionreplay.internal.recorder.MappingContext
+import com.datadog.android.sessionreplay.internal.recorder.base64.ImageWireframeHelper
+import com.datadog.android.sessionreplay.internal.recorder.base64.ImageWireframeHelperCallback
+import com.datadog.android.sessionreplay.internal.recorder.densityNormalized
+import com.datadog.android.sessionreplay.model.MobileSegment
+import com.datadog.android.sessionreplay.utils.StringUtils
+import com.datadog.android.sessionreplay.utils.UniqueIdentifierGenerator
+import com.datadog.android.sessionreplay.utils.ViewUtils
+
+@Suppress("UndocumentedPublicClass")
+abstract class BaseAsyncBackgroundWireframeMapper<T : View>(
+    stringUtils: StringUtils = StringUtils,
+    viewUtils: ViewUtils = ViewUtils
+) : BaseWireframeMapper<T, MobileSegment.Wireframe>(stringUtils, viewUtils) {
+
+    // Why is this nullable ???
+    // TODO: RUM-0000 Make the ImageWireframeHelper non nullable
+    private var imageWireframeHelper: ImageWireframeHelper? = null
+    private var uniqueIdentifierGenerator = UniqueIdentifierGenerator
+
+    internal constructor(
+        imageWireframeHelper: ImageWireframeHelper,
+        uniqueIdentifierGenerator: UniqueIdentifierGenerator
+    ) : this() {
+        this.imageWireframeHelper = imageWireframeHelper
+        this.uniqueIdentifierGenerator = uniqueIdentifierGenerator
+    }
+
+    /**
+     * Maps the [View] into a list of [MobileSegment.Wireframe].
+     */
+    override fun map(
+        view: T,
+        mappingContext: MappingContext,
+        asyncJobStatusCallback: AsyncJobStatusCallback
+    ):
```

Comment:
we need to add a good documentation for this method in the future, because it is not clear why do we have such API - we return some result instantly and at the same time have an async callback with some notification mechanism. Usually APIs have either of the two - return result in a blocking way, or give it in the callback.

Reply:
I do agree, maybe once we will have a stable version, meaning the new resources endpoint in please (which will remove the base64 logic) and the fine grained masking we could simplify a bit this part. My plan is to assess after if we still need this queue or not. We already copy the drawable now so I am thinking to actually create this `ViewPlacehdolder` model that will be processed into wireframe at the `Processor` level. For now we are not ready for this switch as we do not know if copying the drawable will always be possible and also need to see how the logic will look like with the new resources endpoint in place.